nntrainer/src/tensor.cpp

```diff
@@ -770,4 +768,58 @@ Tensor Tensor::standardization() const {
 
   return result;
 }
+
+ExternalLoopInfo Tensor::computeExternalLoop(const Tensor &m) {
+  if (m.length() > this->length())
+    throw exception::not_supported("broadcasting *this is not supported");
+
+  const TensorDim m_dim = m.getDim();
+
+  ExternalLoopInfo e;
+
+  for (unsigned int i = 0; i < MAXDIM; ++i) {
+    if (m_dim.getTensorDim(i) == 1) {
+      continue;
+    }
+
+    if (dim.getTensorDim(i) == m_dim.getTensorDim(i)) {
+      e.strides[i] = m.strides[i];
+      continue;
+    }
+
+    std::stringstream ss;
+    ss << "[computeExternalLoop] broadcasting only allowed for"
+          "dimension value of 1 \n"
+       << "this: " << dim << "target: " << m_dim;
+    throw std::invalid_argument(ss.str().c_str());
+  }
+
+  /// calculate inner loop size
+  unsigned int inner_loop_size = dim.getTensorDim(3);
+
+  bool first_dim_none = m_dim.getTensorDim(3) == 1;
+  e.buffer_axis = -1;
+  for (int axis = 2; axis >= 0; --axis) {
+    unsigned int cur_dim = m_dim.getTensorDim(axis);
+
+    if (first_dim_none) {
+      if (cur_dim != 1) {
+        e.buffer_axis = axis;
```

Comment:
cant we simply find the buffer_axis by the first axis from right where the stride for m and this doesn't match?

Reply:
as discussed offline, lets keep this for now.